rio_tiler/io/cogeo.py

```diff
@@ -685,3 +693,51 @@ def __attrs_post_init__(self):
             )
         )
         super().__attrs_post_init__()
+
+
+@attr.s
+class LocalTileMatrixSet:
+    """Fake TMS for local image."""
+
+    width: int = attr.ib()
+    height: int = attr.ib()
+    tile_size: int = attr.ib(default=256)
+    rasterio_crs: CRS = attr.ib(default=None)
+
+    _minzoom: int = attr.ib(init=False, default=0)
+    _maxzoom: int = attr.ib(init=False)
+
+    def __attrs_post_init__(self):
+        """Get MaxZoom from the dataset."""
+        self._maxzoom = get_maximum_overview_level(
+            self.width,
+            self.height,
+            minsize=self.tile_size,
+        )
+
+    @property
+    def minzoom(self):
+        """Return tms minzoom."""
+        return self._minzoom
+
+    @property
+    def maxzoom(self):
+        """Return tms maxzoom."""
+        return self._maxzoom
+
+    def _ul(self, *tile: morecantile.Tile) -> morecantile.Coords:
+        """Return the upper left coordinate of the (x, y, z) tile."""
+        t = _parse_tile_arg(*tile)
+
+        res = 2.0 ** (self.maxzoom - t.z)
+        xcoord = self.tile_size * t.x * res
+        ycoord = self.tile_size * t.y * res
+
+        return morecantile.Coords(xcoord, ycoord)
+
+    def xy_bounds(self, *tile: morecantile.Tile) -> morecantile.BoundingBox:
+        """Return the bounding box of the (x, y, z) tile"""
+        t = _parse_tile_arg(*tile)
+        left, bottom = self._ul(t)
+        right, top = self._ul(morecantile.Tile(t.x + 1, t.y + 1, t.z))
+        return morecantile.BoundingBox(left, top, right, bottom)
```

Comment:
Origin of the non-geo image is bottom left so tile bounds is (minx, maxy, maxx, many)